python/housinginsights/ingestion/insertion_functions.py

```diff
@@ -0,0 +1,299 @@
+one file into another file.
+"""
+
+import os
+import sys
+from uuid import uuid4
+import pandas as pd
+from datetime import datetime
+import numpy as np
+
+# setup some useful absolute paths
+PYTHON_PATH = os.path.abspath(os.path.join(os.path.dirname(__file__),
+                                           os.pardir, os.pardir))
+DATA_RAW_PATH = os.path.join(PYTHON_PATH, os.pardir, 'data', 'raw')
+
+# add to python system path
+sys.path.append(PYTHON_PATH)
+
+from python.housinginsights.sources.mar import MarApiConn
+from python.housinginsights.sources.models.mar import FIELDS as MAR_FIELDS
+from python.housinginsights.sources.models.DCHousing import PROJECT_FIELDS_MAP,\
+    SUBSIDY_FIELDS_MAP
+from python.housinginsights.sources.models.pres_cat import PROJ_FIELDS, \
+    SUBSIDY_FIELDS
+
+
+def project_csv_fix_empty_address_id():
+    """
+    Identifies any rows in the project.csv file with missing 
+    'proj_address_id' and attempts to fix by using latitude/longitude, 
+    proj_x/y, or proj_addre/city/st/zip.
+    
+    :return: 
+    """
+    proj_df = pd.read_csv(os.path.join(DATA_RAW_PATH, 'preservation_catalog',
+                                       '20170315', 'Project.csv'))
+
+    # get all indices with empty related address fields
+    aid_empty_idx = proj_df[proj_df.Proj_address_id.isnull()].index
+    lat_empty_idx = proj_df[proj_df.Proj_lat.isnull()].index
+    lon_empty_idx = proj_df[proj_df.Proj_lon.isnull()].index
+    x_empty_idx = proj_df[proj_df.Proj_x.isnull()].index
+    y_empty_idx = proj_df[proj_df.Proj_y.isnull()].index
+    address_empty_idx = proj_df[proj_df.Proj_addre.isnull()].index
+
+    mar_api = MarApiConn()
+
+    # iterate through each row and use mar api to get address if possible
+    for idx in aid_empty_idx:
+        if idx not in lat_empty_idx and idx not in lon_empty_idx:
+            result = mar_api.reverse_lat_lng_geocode(proj_df.ix[idx,
+                                                                'Proj_lat'],
+                                                     proj_df.ix[idx,
+                                                                'Proj_lon'])
+        elif idx not in x_empty_idx and idx not in y_empty_idx:
+            result = mar_api.reverse_geocode(proj_df.ix[idx, 'Proj_x'],
+                                             proj_df.ix[idx, 'Proj_y'])
+        elif idx not in address_empty_idx:
+            # check whether address is valid - it has street number
+            try:
+                first_address = proj_df.ix[idx, 'Proj_addre']
+                str_num = first_address.split(' ')[0]
+                int(str_num)
+                result = mar_api.find_location(first_address)
+            except ValueError:
+                result = None
+
+        if result is not None:
+            proj_df.ix[idx, 'Proj_address_id'] = \
+                result['Table1'][0]["ADDRESS_ID"]
+
+    return proj_df
+
+
+def create_mar_csv(project_df):
```

Comment:
We talked about switching this behavior to use the database and adding the appropriate column to the project table in the database. What do you think about when we should do that? Better to deploy using your csv file for now and adjust later? Or should we try to put in that refactor now?